Core/src/TrackFitting/MbfSmoother.cpp

```diff
@@ -33,44 +33,54 @@ void MbfSmoother::visitMeasurement(const InternalTrackState& ts,
                                    BoundVector& small_lambda_hat) const {
   assert(ts.measurement.has_value());
 
-  const auto& measurement = ts.measurement.value();
+  const InternalTrackState::Measurement& measurement = ts.measurement.value();
+  const Acts::BoundMatrix& F = ts.jacobian;
 
   visit_measurement(measurement.calibratedSize, [&](auto N) -> void {
     constexpr std::size_t kMeasurementSize = decltype(N)::value;
 
+    using MeasurementMatrix =
+        Eigen::Matrix<ActsScalar, kMeasurementSize, eBoundSize>;
+    using CovarianceMatrix =
+        Eigen::Matrix<ActsScalar, kMeasurementSize, kMeasurementSize>;
+    using KalmanGainMatrix =
+        Eigen::Matrix<ActsScalar, eBoundSize, kMeasurementSize>;
+
     typename TrackStateTraits<kMeasurementSize, true>::Calibrated calibrated{
         measurement.calibrated};
     typename TrackStateTraits<kMeasurementSize, true>::CalibratedCovariance
         calibratedCovariance{measurement.calibratedCovariance};
 
     // Measurement matrix
-    const auto H = measurement.projector
-                       .template topLeftCorner<kMeasurementSize, eBoundSize>()
-                       .eval();
+    const MeasurementMatrix H =
+        measurement.projector
+            .template topLeftCorner<kMeasurementSize, eBoundSize>()
+            .eval();
```

Comment:
I think if you spell out the type on the left, you don't even need the `eval` anymore. This applies to a number of other spots as well.

Reply:
hopefully done